src/ResourceManager/Sql/Commands.Sql/SqlDatabaseAgent/Services/JobCredential/AzureSqlDatabaseAgentJobCredentialAdapter.cs

```diff
@@ -0,0 +1,157 @@
+        {
+            var resp = Communicator.List(resourceGroupName, serverName, agentName);
+            return resp.Select(credentialName => CreateAgentCredentialModelFromResponse(resourceGroupName, serverName, agentName, credentialName)).ToList();
+        }
+
+        /// <summary>
+        /// Deletes an Azure SQL Database Agent associated to a server
+        /// </summary>
+        /// <param name="resourceGroupName">The resource group name</param>
+        /// <param name="serverName">The server the agent is in</param>
+        /// <param name="agentName">The agent name</param>
+        public void RemoveJobCredential(string resourceGroupName, string serverName, string agentName, string credentialName)
+        {
+            Communicator.Remove(resourceGroupName, serverName, agentName, credentialName);
+        }
+
+        /// <summary>
+        /// Convert a Management.Sql.Models.JobAgent to AzureSqlDatabaseAgentJobCredentialModel
+        /// </summary>
+        /// <param name="resourceGroupName">The resource group the server is in</param>
+        /// <param name="serverName">The server the agent is in</param>
+        /// <param name="resp">The management client server response to convert</param>
+        /// <returns>The converted agent model</returns>
+        private static AzureSqlDatabaseAgentJobCredentialModel CreateAgentCredentialModelFromResponse(string resourceGroupName, string serverName, string agentName, Management.Sql.Models.JobCredential resp)
+        {
+            // Parse credential name from id
+            // This is not expected to ever fail, but in case we have a bug here it's better to provide a more detailed error message
+            int lastSlashIndex = resp.Id.LastIndexOf('/');
+            string credentialName = resp.Id.Substring(lastSlashIndex + 1);
+
+            AzureSqlDatabaseAgentJobCredentialModel credential = new AzureSqlDatabaseAgentJobCredentialModel
+            {
+                ResourceGroupName = resourceGroupName,
+                ServerName = serverName,
+                AgentName = agentName,
+                CredentialName = resp.Name,
+                UserName = resp.Username,
+                ResourceId = resp.Id
+            };
+
+            return credential;
+        }
+
+        /// <summary>
+        /// Convert a <see cref="SecureString"/> to a plain-text string representation.
+        /// This should only be used in a proetected context, and must be done in the same logon and process context
+        /// in which the <see cref="SecureString"/> was constructed.
+        /// </summary>
+        /// <param name="secureString">The encrypted <see cref="SecureString"/>.</param>
+        /// <returns>The plain-text string representation.</returns>
+        [PermissionSet(SecurityAction.Demand, Name = "FullTrust")]
+        internal static string Decrypt(SecureString secureString)
```

Comment:
Please use this instead: https://github.com/Azure/azure-powershell/blob/preview/src/Common/Commands.Common/Utilities/ConversionUtilities.cs#L95